src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,272 @@
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.walker.EdgeWalker;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * A RoadNameSpellingConsistencyCheckWalker can be used to collect all edges that have NameTag
+ * values that are at some Levenshtein distance (configurable) from the starting edge's NameTag
+ * value. Collected edges are within some linear search area (configurable).
+ *
+ * @author seancoulter
+ */
+class RoadNameSpellingConsistencyCheckWalker extends EdgeWalker
+{
+
+    /**
+     * Directional character, sometimes appended to start/end of road name (e.g. Banana Ave W)
+     */
+    enum Direction
+    {
+        N,
+        S,
+        E,
+        W
+    }
+
+    private static final EnumSet<Direction> DIRECTIONS = EnumSet.of(Direction.N, Direction.S,
+            Direction.E, Direction.W);
+
+    private static final int NINE = 9;
+
+    private static final int ZERO = 0;
+
+    // ASCII character to digit offset
+    private static final int ASCII_OFFSET = 48;
+
+    /**
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumAllowedDifferences
+     *            the number of Levenshtein edits allowed
+     * @return edges that are in the search area and whose names are at most the desired Levenshtein
+     *         distance from the start edge's
+     */
+    static Predicate<Edge> getCandidateEdges(final Edge startEdge,
+            final int maximumAllowedDifferences)
+    {
+        // evaluate name tag of startEdge vs incoming Edge
+        return incomingEdge ->
+        {
+            final int similarityIndex = similarityIndex(incomingEdge, startEdge);
+            return similarityIndex <= maximumAllowedDifferences && similarityIndex > 0;
+        };
+    }
+
+    /**
+     * @param first
+     *            a character to examine for equality with second
+     * @param second
+     *            a character to examine for equality with first
+     * @return 0 if first & second are equal, 1 otherwise
+     */
+    private static int costOfSubstitution(final char first, final char second)
+    {
+        return first == second ? 0 : 1;
+    }
+
+    /**
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumSearchDistance
+     *            the maximum distance from the end of the incoming edge to the start of the
+     *            starting edge
+     * @return A stream of edges that fall in the search area
+     */
+    private static Function<Edge, Stream<Edge>> edgesWithinMaximumSearchDistance(
+            final Edge startEdge, final Distance maximumSearchDistance)
+    {
+        return incomingEdge -> incomingEdge.end().getLocation()
+                .distanceTo(startEdge.start().getLocation())
```

Comment:
It seems a bit odd to me to use an EdgeWalker to try to find things in an area. It would make more sense to use linear or network distance with an EdgeWalker, I think. If for this check it makes sense to use area, then we should be using the atlas spatial indices and do something like `atlas.edgesWithin(start.bounds().expand(maximumSearchDistance))`

Reply:
Thanks for the suggestion--I didn't know about this approach. I think it's a great idea, but before I make any major changes to the search logic I'd like to point out one thing. The current EdgeWalker approach ensures that collected edges are part of the same road network, meaning we can guarantee that all collected edges have a common connection in the ```startEdge```. Using spatial indices seems like it raises the possibility of collecting edges that aren't connected to the ```startEdge```, or are only connected through a ton of degrees of separation, which I believe means they probably aren't from the same road.

Do you have any thoughts about this?

Reply:
I see what you mean, that does make sense. It still feels a bit weird, but I don't have a better solution at the moment.